paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -264,7 +265,7 @@ internal class DefaultFlowController @Inject internal constructor(
         initData: InitData
     ) {
         val confirmParamsFactory =
-            ConfirmStripeIntentParamsFactory.createFactory(initData.clientSecret)
+            ConfirmStripeIntentParamsFactory.createFactory(initData.preferredClientSecret())
```

Comment:
[iOS](https://github.com/stripe-ios/stripe-ios/blob/a0cf5410a9b15866d7565ae3479aeaf55c68cd49/Stripe/PaymentSheet+Helpers.swift#L102).  This difference took a while to track down.  Note how the `clientSecret` used is not the initial one passed in, but from the intent retrieved.

Reply:
hmm, how did this ever work?  Will getTypedClientSecret ever return null?

Reply:
Normally, `initData.clientSecret` is the intent's client secret (`pi_aaa_secret_bbb`) so the retrieved value should be the same.  However, we're adding support for passing in the intent ID (`pi_aaa`) for `initData.clientSecret`.

This change makes it so that, after retrieving the intent, the rest of the flow has `clientSecret` as the actual intent client secret (`pi_aaa_secret_bbb`).

Reply:
Removed this capability, as discussed offline.